### DISCUSSION TITLE
Repeatedly rendered results

### DISCUSSION THREAD

**Discussion by kursame** [KaTeX/KaTeX#2806] [Q&A] [answered]
Hi, some unexpected results appears while using some plugins using KaTex
Input: c = \\pm\\sqrt{a^2 + b^2}
Output: 
<img width="230" alt="httpswww 5axxw comtoolswebweb_run html" src="https://user-images.githubusercontent.com/62455758/108811663-98b82700-75e8-11eb-9de4-dca4ead2539b.png">
(Extra sqrt and some letters)

I tried markdown-it, remark and showdown and the plugins using KaTex. The output are the same.

I don’t know what’s wrong and how to handle it. Thanks in advance.

**Comment by edemaine** [comment, marked as answer]
You should look at the DOM via Inspect Element to see what's going on.

KaTeX renders the input twice by default, once in HTML and one in MathML.  The MathML is normally hidden.  My guess would be that you've changed the CSS (or not included KaTeX's CSS file) so the MathML is rendering too.

**Reply by kursame** [reply]
Oh!
It works
Thank you!
😆

**Reply by kreslotim** [reply]
It works for me too, after 3 hours of torment with chatGPT xD. Thank you!

**Reply by IlIIIIIIlI** [reply]
it works for me too, after samely 3 hours battle with chatGPT. lmao... how can I miss the ```import 'katex/dist/katex.min.css';```...

**Reply by shubhjhawar** [reply]
can someone please share the katex css file link

**Reply by Pushpajekumar** [reply]
`import 'katex/dist/katex.min.css';` This is css file

**Comment by roman-smolnyk** [comment]
For someone like me, be aware of overriding span component in react markdown. `className` is important

```
components={{
    span({ node, className, ...props }) {
        // className should be passed!!!
        return <span {...props} className={className} />;
    }
}}
```
